surfnperf.js

```diff
@@ -356,6 +356,56 @@
     return this.duration('navigationStart', 'loadEventEnd');
   };
 
+  /**
+   * Time to first paint (browser-reported time for the first pixels to be rendered on the page)
+   *
+   * @returns {integer} time in ms
+   * @memberOf SurfNPerf
+   */
+  SNPProto.getFirstPaint = function() {
+    if(this.chromeLoadTimes()) {
+      return this.chromeLoadTimes().firstPaintTime * 1000 - this.navigationStartTiming();
+    } else if(window.performance && window.performance.timing && this.msFirstPaint()) {
+      return this.msFirstPaint() - this.navigationStartTiming();
+    } else {
+      return null;
+    }
+  };
+
+  /**
+   * Time to first paint (Time calculated based on window.requestAnimationFrame() for the first pixels to be rendered on the page)
+   *
+   * @returns {integer} time in ms
+   * @memberOf SurfNPerf
+   */
+  SNPProto.getFirstPaintFrame = function(options) {
+    return this.duration('navigationStart', 'firstPaintFrame', options);
+  };
+
+  SNPProto.chromeLoadTimes = function() {
+    if(window.chrome && window.chrome.loadTimes()) {
+      return window.chrome.loadTimes();
+    } else {
+      return undefined;
+    }
+  };
+
+  SNPProto.msFirstPaint = function() {
+    if(window.performance.timing.msFirstPaint) {
+      return window.performance.timing.msFirstPaint;
+    } else {
+      return undefined;
+    }
+  };
+
+  SNPProto.navigationStartTiming = function() {
```

Comment:
You don't need this function. Spying on the existing `SurfNPerf.performanceTiming()` function will be sufficient for your tests